src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,81 @@
-export const Pagination = () => {};
+import React from 'react';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+function getButtons(total: number, perPage: number) {
+  const numberButtons = Math.ceil(total / perPage);
+  const arr: number[] = [];
+
+  for (let i = 1; i <= numberButtons; i++) {
+    arr.push(i);
+  }
+
+  return arr;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  return (
+    <ul className="pagination">
+      <li className="page-item disabled">
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled="true"
+          data-page={currentPage - 1}
+          onClick={event =>
+            onPageChange(
+              parseInt((event.target as HTMLAnchorElement).dataset.page ?? '1'),
+            )
+          }
```

Comment:
Try to simplify this flow, because right now you have function inside function inside function

Create a handler function with variables